src/Tools/dotnet-counters/CounterPayload.cs

```diff
@@ -50,12 +50,16 @@ class IncrementingCounterPayload : ICounterPayload
         public string m_Value;
         public string m_DisplayName;
         public string m_DisplayRateTimeScale;
-        public IncrementingCounterPayload(IDictionary<string, object> payloadFields)
+        public IncrementingCounterPayload(IDictionary<string, object> payloadFields, int interval)
         {
             m_Name = payloadFields["Name"].ToString();
             m_Value = payloadFields["Increment"].ToString();
             m_DisplayName = payloadFields["DisplayName"].ToString();
-            m_DisplayRateTimeScale = TimeSpan.Parse(payloadFields["DisplayRateTimeScale"].ToString()).ToString("%s' sec'");
+            m_DisplayRateTimeScale = payloadFields["DisplayRateTimeScale"].ToString();
+
+            // In case these properties are not provided, set them to appropriate values.
+            m_DisplayName = m_DisplayName.Length == 0 ? m_Name : m_DisplayName;
+            m_DisplayRateTimeScale = m_DisplayRateTimeScale.Length == 0 ? $"{interval} sec" : m_DisplayRateTimeScale;
```

Comment:
Do you still want to use `TimeSpan.Parse(m_DisplayRateTimeScale).ToString("%s sec")` on the right side of the ternary operator?

Reply:
Yes I do. Thanks for the catch :)